src/net.cpp

```diff
@@ -437,21 +451,22 @@ static CAddress GetBindAddress(const Sock& sock)
     return addr_bind;
 }
 
-CNode* CConnman::ConnectNode(CAddress addrConnect, const char *pszDest, bool fCountFailure, ConnectionType conn_type)
+std::pair<CNode*, ServiceFlags> CConnman::ConnectNode(CAddress addrConnect, const char *pszDest, bool fCountFailure, ConnectionType conn_type)
 {
     AssertLockNotHeld(m_unused_i2p_sessions_mutex);
     assert(conn_type != ConnectionType::INBOUND);
+    ServiceFlags node_services = nLocalServices;
 
     if (pszDest == nullptr) {
         if (IsLocal(addrConnect))
-            return nullptr;
+            return std::make_pair(nullptr, node_services);
```

Comment:
nit: Dislike scattering this all over the place. `node_services` doesn't even have a meaning with nullptr. Maybe return `std::optional<...>` instead?

Reply:
Seems more elegant.